examples/connext_dds/compression/c++/compression_publisher.cxx

```diff
@@ -0,0 +1,238 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the software solely for use with RTI Connext DDS. Licensee may
+* redistribute copies of the software provided that all such copies are subject
+* to this license. The software is provided "as is", with no warranty of any
+* type, including any warranty for fitness for any purpose. RTI is under no
+* obligation to maintain or support the software. RTI shall not be liable for
+* any incidental or consequential damages arising out of the use or inability
+* to use the software.
+*/
+
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <vector>
+
+#include "compression.h"
+#include "compressionSupport.h"
+#include "ndds/ndds_cpp.h"
+#include "application.h"
+
+using namespace application;
+
+static int shutdown_participant(
+    DDSDomainParticipant *participant,
+    const char *shutdown_message,
+    int status);
+
+int run_publisher_application(
+        unsigned int domain_id,
+        unsigned int sample_count,
+        std::string compression_id,
+        std::string input_file)
+{
+    // Start communicating in a domain, usually one participant per application
+    DDSDomainParticipant *participant =
+    DDSTheParticipantFactory->create_participant(
+        domain_id,
+        DDS_PARTICIPANT_QOS_DEFAULT,
+        NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
```

Comment:
Take a look at indentation, I think CI will flag all this.

Reply:
I left the indentation as the code was generated. The CI didn't flag it, but I will change it anyway.